nexus-restore-r/src/main/java/org/sonatype/nexus/blobstore/restore/r/internal/RRestoreBlobStrategy.java

```diff
@@ -0,0 +1,139 @@
+import static org.eclipse.aether.util.StringUtils.isEmpty;
+import static org.sonatype.nexus.common.hash.HashAlgorithm.SHA1;
+
+/**
+ * @since 1.1.next
+ */
+@Named(RFormat.NAME)
+@Singleton
+public class RRestoreBlobStrategy
+    extends BaseRestoreBlobStrategy<RRestoreBlobData>
+{
+  @Inject
+  public RRestoreBlobStrategy(final NodeAccess nodeAccess,
+                              final RepositoryManager repositoryManager,
+                              final BlobStoreManager blobStoreManager,
+                              final DryRunPrefix dryRunPrefix)
+  {
+    super(nodeAccess, repositoryManager, blobStoreManager, dryRunPrefix);
+  }
+
+  @Override
+  protected RRestoreBlobData createRestoreData(final RestoreBlobData restoreBlobData) {
+    checkState(!isEmpty(restoreBlobData.getBlobName()), "Blob name cannot be empty");
+
+    return new RRestoreBlobData(restoreBlobData);
+  }
+
+  @Override
+  protected boolean canAttemptRestore(@Nonnull final RRestoreBlobData rRestoreBlobData) {
+    Repository repository = getRepository(rRestoreBlobData);
+    Optional<RRestoreFacet> rRestoreFacetFacet = repository.optionalFacet(RRestoreFacet.class);
+
+    if (!rRestoreFacetFacet.isPresent()) {
+      log.warn("Skipping as R Restore Facet not found on repository: {}", repository.getName());
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  protected String getAssetPath(@Nonnull final RRestoreBlobData rRestoreBlobData) {
+    return rRestoreBlobData.getBlobData().getBlobName();
+  }
+
+  @Override
+  protected boolean assetExists(@Nonnull final RRestoreBlobData rRestoreBlobData) {
+    RRestoreFacet facet = getRestoreFacet(rRestoreBlobData);
+
+    return facet.assetExists(getAssetPath(rRestoreBlobData));
+  }
+
+  @Override
+  protected void createAssetFromBlob(@Nonnull final AssetBlob assetBlob,
+                                     @Nonnull final RRestoreBlobData rRestoreBlobData)
+      throws IOException
+  {
+    RRestoreFacet facet = getRestoreFacet(rRestoreBlobData);
+    final String path = rRestoreBlobData.getBlobData().getBlobName();
+
+    facet.restore(assetBlob, path);
+  }
+
+  @Nonnull
+  @Override
+  protected List<HashAlgorithm> getHashAlgorithms() {
+    return ImmutableList.of(SHA1);
+  }
+
+  @Override
+  protected boolean componentRequired(final RRestoreBlobData data) {
+    RRestoreFacet facet = getRestoreFacet(data);
+    final String path = data.getBlobData().getBlobName();
+
+    return facet.componentRequired(path);
+  }
+
+  @Override
+  protected Query getComponentQuery(final RRestoreBlobData data) throws IOException {
+    RRestoreFacet facet = getRestoreFacet(data);
+    RestoreBlobData blobData = data.getBlobData();
+    Map<String, String> attributes;
+    try (InputStream inputStream = blobData.getBlob().getInputStream()) {
+      attributes = facet.extractComponentAttributesFromArchive(blobData.getBlobName(), inputStream);
+    }
+
+    return facet.getComponentQuery(attributes);
```

Comment:
Question: Is `attributes` guaranteed to have the attributes needed for the query?

Reply:
Should have, because we use the same 'extractComponentnAttributes' when we store archive